nixos/modules/security/pam.nix

```diff
@@ -337,6 +345,8 @@ let
         (''
           # Account management.
           account required pam_unix.so
+          ${optionalString cfg.enableHomed
+            "-account [success=1 default=ignore] ${pkgs.systemd}/lib/security/pam_systemd_home.so"}
```

Comment:
This makes `pam_acct_mgmt()` equivalent to ```int pam_acct_mgmt() {
    return PAM_SUCCESS;
}```
We have the same problem in Arch. I would suggest either patching systemd or following https://github.com/systemd/systemd/issues/16906#issuecomment-692088553 for minimum (one) loopholes.

Reply:
Can you give me an example of how to configure this? I'm not that good with pam

Reply:
> I would suggest either patching systemd or following [systemd/systemd#16906 (comment)](https://github.com/systemd/systemd/issues/16906#issuecomment-692088553) for minimum (one) loopholes.

patching systemd would be just removing [L836](https://github.com/systemd/systemd/blob/092f5d2ed34e0a72995cd6808ba7bf1c535561ec/src/home/pam_systemd_home.c#L836) and [L837](https://github.com/systemd/systemd/blob/092f5d2ed34e0a72995cd6808ba7bf1c535561ec/src/home/pam_systemd_home.c#L837). Then you can follow https://github.com/systemd/systemd/issues/16906#issuecomment-683439776 (this is a different comment from the one quoted above).

Reply:
I'd suggest awaiting the outcome of https://github.com/systemd/systemd/issues/16906 - this should really be fixed upstream.

Reply:
Also, our own PAM infrastructure is somewhat of a mess currently: https://github.com/NixOS/nixpkgs/issues/90640

Reply:
Apparently, this should have been fixed in https://github.com/systemd/systemd/pull/17376, which should have landed in systemd 247.

Reply:
The PAM refactor is ongoing in https://github.com/NixOS/nixpkgs/pull/105319.